<!DOCTYPE html>
<html lang="en"><head>
<script src="slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="slides_files/libs/quarto-html/popper.min.js"></script>
<script src="slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="slides_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="slides_files/libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="slides_files/libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="slides_files/libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.41">

  <meta name="author" content="James Mba Azam, PhD">
  <meta name="dcterms.date" content="2024-06-09">
  <title>Introduction to Infectious Disease Modelling</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
      margin-bottom: 0em;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/theme/quarto.css">
  <link href="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Introduction to Infectious Disease Modelling</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
James Mba Azam, PhD <a href="https://orcid.org/0000-0001-5782-7330" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a>
</div>
<div class="quarto-title-author-email">
<a href="mailto:james.azam@lshtm.ac.uk">james.azam@lshtm.ac.uk</a>
</div>
        <p class="quarto-title-affiliation">
            Epiverse-TRACE Initiative, London School of Hygiene and Tropical Medicine, UK
          </p>
    </div>
</div>

  <p class="date">2024-06-09</p>
</section><section id="TOC">
<nav role="doc-toc"> 
<h2 id="toc-title">Table of contents</h2>
<ul>
<li><a href="#/an-overview-of-infectious-diseases" id="/toc-an-overview-of-infectious-diseases">An overview of infectious diseases</a></li>
<li><a href="#/infectious-disease-models" id="/toc-infectious-disease-models">Infectious disease models</a></li>
<li><a href="#/introduction-to-compartmental-models" id="/toc-introduction-to-compartmental-models">Introduction to compartmental models</a></li>
<li><a href="#/controlling-epidemics" id="/toc-controlling-epidemics">Controlling epidemics</a></li>
<li><a href="#/an-overview-of-host-heterogeneity" id="/toc-an-overview-of-host-heterogeneity">An overview of host heterogeneity</a></li>
<li><a href="#/wrap-up-and-qa" id="/toc-wrap-up-and-qa">Wrap-Up and Q&amp;A</a></li>
<li><a href="#/list-of-resources" id="/toc-list-of-resources">List of Resources</a></li>
</ul>
</nav>
</section>
<section>
<section id="an-overview-of-infectious-diseases" class="title-slide slide level1 center">
<h1>An overview of infectious diseases</h1>

</section>
<section id="what-are-infectious-diseasess" class="slide level2">
<h2>What are infectious diseasess</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/disease_definition.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img data-src="images/disease_definition.png" class="quarto-figure quarto-figure-center" data-fig-cap="Definition of a disease according to the Merriam Webster dictionary"></a></p>
</figure>
</div>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:45%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/rotavirus.avif" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="A 3D graphical representation of Rotavirus virions."><img data-src="images/rotavirus.avif" alt="A 3D graphical representation of Rotavirus virions."></a></p>
<figcaption>A 3D graphical representation of Rotavirus virions.</figcaption>
</figure>
</div>
</div><div class="column" style="width:55%;">
<p>Diseases can be classified according to:</p>
<ul>
<li><span style="color:tomato">Cause</span> (e.g., infectious, non-infectious)</li>
<li><span style="color:tomato">Duration</span> (e.g., acute, chronic)</li>
<li><span style="color:tomato">Mode of transmission</span> (direct or indirect)</li>
<li><span style="color:tomato">Impact</span> on the host (e.g., fatal, non-fatal)</li>
</ul>
</div>
</div>
</section>
<section class="slide level2">

<p>Note that these classifications are not mutually exclusive. Hence, a disease can be classified under more than one category at a time.</p>
</section>
<section id="sec-control-measures" class="slide level2">
<h2>How are infectious diseases controlled?</h2>
<ul>
<li>In general, infectious disease control aims to reduce disease transmission.</li>
<li>The type of control used depends on the disease and its characteristics.</li>
<li>Broadly, there are two main types of control measures:
<ul>
<li>Pharmaceutical interventions (PIs)</li>
<li>Non-pharmaceutical interventions (NPIs)</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<h3 id="pharmaceutical-interventions-pis">Pharmaceutical interventions (PIs)</h3>
<ul>
<li><p>Pharmaceutical Interventions are medical interventions that target the pathogen or the host.</p></li>
<li><p>Examples:</p>
<ul>
<li>Vaccines,</li>
<li>Antiviral drugs, and</li>
<li>Antibiotics.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<h4 id="sec-vaccination">Vaccination</h4>
<ul>
<li>Vaccination is the effective way to prevent infectious diseases.</li>
<li>Activates the host’s immune system to produce antibodies against the pathogen.</li>
<li>It is generally applied prophylactically to susceptible individuals (before infection); this reduces the risk of infection and disease.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Challenges with vaccination:
<ul>
<li>Take time to develop for new pathogens</li>
<li>Never 100% effective and limited duration of protection</li>
<li>Adverse side effects</li>
<li>Some individuals cannot be vaccinated or refuce vaccination</li>
<li>Some pathogens mutate rapidly (e.g., influenza virus)</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<h3 id="sec-npi">Non-pharmaceutical interventions (NPIs)</h3>
<ul>
<li><p>Non-pharmaceutical interventions are measures that do not involve medical interventions.</p></li>
<li><p>Examples:</p></li>
<li><p>Quarantine,</p></li>
<li><p>Physical/social distancing, and</p></li>
<li><p>Mask-wearing.</p></li>
</ul>
</section>
<section class="slide level2">

<h4 id="quarantine">Quarantine</h4>
<ul>
<li><em>Isolation of individuals who may have been exposed to a contagious disease</em>.</li>
<li>Advantage is that it’s simple and its effectiveness does not depend on the disease.</li>
<li>Disadvantages include:
<ul>
<li>Infringement on individual rights</li>
<li>Can be difficult to enforce</li>
<li>Can be costly</li>
<li>Can lead to social stigma</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<h4 id="contact-tracing">Contact tracing</h4>
<ul>
<li>Contact tracing is used to identify exposed individuals, i.e., individuals who might been in contact with an infected/infectious individual.</li>
<li>It involves identifying, assessing, and managing people who have been exposed to a contagious disease to prevent further transmission.</li>
<li>It is a critical component of infectious disease surveillance and is often used in combination with other control measures.</li>
</ul>
</section>
<section class="slide level2">

<div class="callout callout-caution no-icon callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<p><strong>Discussion</strong></p>
</div>
<div class="callout-content">
<ul>
<li>How can we quantify the impact of a control measure?</li>
</ul>
</div>
</div>
</div>
</section></section>
<section>
<section id="infectious-disease-models" class="title-slide slide level1 center">
<h1>Infectious disease models</h1>

</section>
<section id="what-are-infectious-disease-models" class="slide level2">
<h2>What are infectious disease models?</h2>
<ul>
<li><span style="color:tomato;"><em>Models</em></span> generally refer to conceptual representations of an object or system.</li>
<li><span style="color:tomato;"><em>Mathematical models</em></span> use mathematics to describe the system. For example, the famous <span class="math inline">\(E = mc^2\)</span> is a mathematical model that describes the relationship between mass and energy.</li>
<li><span style="color:tomato;"><em>Infectious disease models</em></span> use mathematics/statistics to represent dynamics/spread of infectious diseases.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Mathematical models can be used to link the biological process of disease transmission and the emergent dynamics of infection at the population level.</li>
<li>Models require making some assumptions and abstractions.</li>
</ul>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:60%;">
<ul>
<li>By definition, “all models are wrong, but some are useful” <span class="citation" data-cites="Box1979">(<a href="#/references" role="doc-biblioref" onclick="">Box 1979</a>)</span>.
<ul>
<li>Good enough models are those that capture the essential features of the system being studied.</li>
</ul></li>
</ul>
</div><div class="column" style="width:40%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/GeorgeEPBox.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="By DavidMCEddy at en.wikipedia, CC BY-SA 3.0, https://commons.wikimedia.org/w/index.php?curid=115167166"><img data-src="images/GeorgeEPBox.jpeg" alt="By DavidMCEddy at en.wikipedia, CC BY-SA 3.0, https://commons.wikimedia.org/w/index.php?curid=115167166"></a></p>
<figcaption>By DavidMCEddy at en.wikipedia, CC BY-SA 3.0, https://commons.wikimedia.org/w/index.php?curid=115167166</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li>What makes models “wrong” by definition?:
<ul>
<li>Simplifications of reality; not capturing all the complexities of the system being studied.</li>
</ul></li>
</ul>
</section>
<section id="factors-that-influence-model-formulationchoice" class="slide level2">
<h2>Factors that influence model formulation/choice</h2>
<ul>
<li>Accuracy: how well does the model to reproduce observed data and predict future outcomes?</li>
<li>Transparency: is it easy to understand and interpret the model and its outputs? (This is affected by the model’s complexity)</li>
<li>Flexibility: the ability of the model to be adapted to different scenarios.</li>
</ul>
</section>
<section id="what-are-models-used-for" class="slide level2">
<h2>What are models used for?</h2>
<ul>
<li>Generally, models can be used to predict and understand/explain the dynamics of infectious diseases.</li>
</ul>
<div class="callout callout-caution no-icon callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<p><strong>Discussion</strong></p>
</div>
<div class="callout-content">
<p>How are these two uses impacted by accuracy, transparency, and flexibility?</p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="prediction-of-the-future-course">Prediction of the future course</h3>
<div class="columns">
<div class="column" style="width:40%;">
<ul>
<li>Must be <span style="color:tomato;">accurate</span>.</li>
<li>“But the estimate proved to be off. Way, way off. Like, 65 times worse than what ended up happening.”</li>
</ul>
</div><div class="column" style="width:60%;">
<p><a href="images/wrong_ebola_deaths_estimate.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img data-src="images/wrong_ebola_deaths_estimate.png" data-fig-cap="[CDC’s top modeler courts controversy with disease estimate](https://apnews.com/domestic-news-domestic-news-fbb4fc8921d54201a1c5ca91e5b601f5)"></a></p>
</div>
</div>
</section>
<section class="slide level2">

<p>&lt; Insert examples of models that have made accurate predictions of the future course of an outbreak &gt;</p>
<ul>
<li>See &lt;Justin Lessler, Derek A. T. Cummings, Mechanistic Models of Infectious Disease and Their Impact on Public Health, American Journal of Epidemiology, Volume 183, Issue 5, 1 March 2016, Pages 415–422, https://doi.org/10.1093/aje/kww021&gt;</li>
</ul>
</section>
<section class="slide level2">

<h3 id="understanding-or-explaining-disease-dynamics">Understanding or explaining disease dynamics</h3>
<ul>
<li>Models can be used to understand how a disease spreads and how its spread can be controlled.</li>
<li>The insights gained from models can be used to:
<ul>
<li>inform public health policy and interventions.</li>
<li>design interventions to control the spread of the disease, for example, randomised controlled trials.</li>
<li>collect new data.</li>
<li>build predictive models.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<p>&lt; Insert examples of models that explain the spread and dynamics of infectious diseases &gt;</p>
<ul>
<li>See &lt;Justin Lessler, Derek A. T. Cummings, Mechanistic Models of Infectious Disease and Their Impact on Public Health, American Journal of Epidemiology, Volume 183, Issue 5, 1 March 2016, Pages 415–422, https://doi.org/10.1093/aje/kww021&gt;</li>
</ul>
</section>
<section class="slide level2">

<p>&lt; Insert examples of models that evaluate the impact of interventions and determine the next course of action &gt;</p>
</section>
<section id="limitations-of-infectious-disease-models" class="slide level2">
<h2>Limitations of infectious disease models</h2>
<ul>
<li>Host behaviour is often difficult to predict.</li>
<li>The pathogen often has unknown characteristics or known characteristics that are difficult to model.</li>
<li>Data is often not available or is of poor quality.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Models:
<ul>
<li>Simplifications of reality and do not capture all the complexities of the system being studied.</li>
<li>Only as good as the data used to parameterize them.</li>
<li>Can be sensitive to the assumptions made during their formulation.</li>
<li>Can be computationally expensive and require a lot of data to run.</li>
<li>Can be difficult to interpret and communicate to non-experts.</li>
</ul></li>
</ul>
</section></section>
<section>
<section id="introduction-to-compartmental-models" class="title-slide slide level1 center">
<h1>Introduction to compartmental models</h1>

</section>
<section id="what-are-compartmental-models" class="slide level2">
<h2>What are compartmental models?</h2>
<ul>
<li>Compartmental models:
<ul>
<li>divide populations into compartments (or groups) based on the individual’s infection status and track them through time <span class="citation" data-cites="Blackwood2018a">(<a href="#/references" role="doc-biblioref" onclick="">Blackwood and Childs 2018</a>)</span>.</li>
<li>are mechanistic, meaning they describe processes such the interaction between hosts, biological processes of pathogen, host immune response, and so forth.</li>
</ul></li>
<li>Compartmental models are different from statistical models, which are used to describe the relationship between variables.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Individuals in a compartment:
<ul>
<li>are assumed to have the same features (disease state, age, location, etc)</li>
<li>can only be in one compartment at a time.</li>
<li>move between compartments based on defined transition rates.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:60%;">
<ul>
<li>Common compartments:
<ul>
<li>Susceptible (S) - hosts are not infected but can be infected</li>
<li>Infected (I) - hosts are infected (and can infect others)</li>
<li>Removed (R) - hosts are no longer infected and cannot be re-infected</li>
</ul></li>
</ul>
</div><div class="column" style="width:40%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/infection_timeline.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Infection timeline illustrating how a pathogen in a host interacts with the host’s immune system (Source: Modelling Infectious Diseases of Humans and Animals)"><img data-src="images/infection_timeline.png" alt="Infection timeline illustrating how a pathogen in a host interacts with the host’s immune system (Source: Modelling Infectious Diseases of Humans and Animals)"></a></p>
<figcaption>Infection timeline illustrating how a pathogen in a host interacts with the host’s immune system (Source: Modelling Infectious Diseases of Humans and Animals)</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li><p>Other compartments can be added to the model to account for important events or processes (e.g., exposed, recovered, vaccinated, etc.)</p></li>
<li><p>It is, however, important to keep the model simple, less computationally intensive, and interpretable.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Compartmental models either have <em>discrete</em> or <em>continuous</em> time scales:
<ul>
<li><span style="color:tomato;">Discrete time scales</span>: time is divided into discrete intervals (e.g., days, weeks, months).</li>
<li><span style="color:tomato;">Continuous time scales</span>: time is continuous and the model is described using differential equations.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Compartmental models can be <em>deterministic</em> or <em>stochastic</em>:
<ul>
<li><span style="color:tomato;">Deterministic</span> models always return the same output for the same input.</li>
<li><span style="color:tomato;">Stochastic</span> models account for randomness in the system and model output always varies. Hence, they are often run multiple times to get an average output.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>The choice of model type depends on the research question, data availability, and computational resources.</p></li>
<li><p>In this introduction, we will focus on <span style="color:tomato;">deterministic compartmental models with continuous time scales</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Now, back to the models, we are going to consider infections that either confer immunity after recovery or not.</li>
<li>The simplest compartmental models for capturing this is the SIR model.</li>
</ul>
</section>
<section id="the-susceptible-infected-recovered-sir-model" class="slide level2">
<h2>The Susceptible-Infected-Recovered (SIR) model</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="./images/model_diagrams/model_diagrams.001.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img data-src="./images/model_diagrams/model_diagrams.001.jpeg" class="quarto-figure quarto-figure-center" style="width:40.0%"></a></p>
</figure>
</div>
<p>This model groups individuals into three <em>disease states</em>:</p>
<ul>
<li><p><span style="color:green;">Susceptible (S)</span>: not infected but can be.</p></li>
<li><p><span style="color:tomato;">Infected (I)</span>: infected &amp; infectious.</p></li>
<li><p><span style="color:blue;">Recovered/removed (R)</span>: recovered &amp; immune.</p></li>
</ul>
</section>
<section class="slide level2">

<h3 id="how-do-individuals-move-between-compartments">How do individuals move between compartments?</h3>
<h4 id="process-1-transmission">Process 1: Transmission</h4>
<p><a href="images/model_diagrams/model_diagrams.002.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img data-src="images/model_diagrams/model_diagrams.002.jpeg" style="width:70.0%"></a></p>
<p><span style="color:red;">What drives transmission?</span></p>
</section>
<section class="slide level2">

<ul>
<li>Transmission is driven by several factors, including:
<ul>
<li>Disease prevalence, <span style="color:blue;"><span class="math inline">\(I\)</span></span>, i.e., number of infected individuals at the time.</li>
<li>The number of contacts, <span style="color:blue;"><span class="math inline">\(C\)</span></span>, susceptible individuals have with infected individual.</li>
<li>The probability, <span style="color:blue;"><span class="math inline">\(p\)</span></span>, a susceptible individual will become infected when they contact an infected individual.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<p><a href="images/model_diagrams/model_diagrams.002.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img data-src="images/model_diagrams/model_diagrams.002.jpeg" style="width:70.0%"></a></p>
<p>The transmission term is often defined through the <span style="color:tomato;">force of infection (FOI), <span class="math inline">\(\lambda\)</span></span>.</p>
</section>
<section class="slide level2">

<ul>
<li>FOI, <span class="math inline">\(\lambda\)</span>, is the <span style="color:tomato;">per capita rate</span> at which susceptible individuals become infected.</li>
</ul>
<div class="callout callout-note no-icon callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>“Per capita” means the rate of an event occurring per individual in the population per unit of time.</p>
</div>
</div>
</div>
<ul>
<li>Given the rate per individual per time, FOI, the rate at which new infecteds are generated is given by <span style="color:blue;"><span class="math inline">\(\lambda \times S\)</span></span>, where <span class="math inline">\(S\)</span> is the number of susceptible individuals.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>The force of infection is made up of three probabilities/rates:
<ul>
<li>the probability/rate that contacts happen,</li>
<li>the probability/rate that a given contact is with an infected individual, and</li>
<li>the probability/rate that a contact results in successful transmission.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>The FOI can be formulated in two ways, depending on how the contact rate is expected to change with the population size:
<ul>
<li>Frequency-dependent/mass action transmission</li>
<li>Density-dependent transmission</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><span style="color:tomato;"><em>Frequency-dependent/mass action transmission</em></span>: The rate of contact between individuals does not depend on the population size. Here, <span style="color:blue;"><span class="math inline">\(\lambda = \beta \times \dfrac{I}{N}\)</span></span>.
<ul>
<li>Often used to model sexually-transmitted diseases and diseases with heterogeneity in contact rates.</li>
<li>Sexual transmission in this case does not depend on how many infected individuals are in the population.</li>
</ul></li>
</ul>
<aside class="notes">
<p>Frequency-dependent transmission assumes the interaction between susceptible and infected individuals does not change with the population size.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<ul>
<li><span style="color:tomato;"><em>Density dependent transmission</em></span>: The rate of contact between individuals is dependent on the population size. Here, <span style="color:blue;"><span class="math inline">\(\lambda = \beta \times I\)</span></span>:
<ul>
<li>Transmission rates increase with population size/density.</li>
</ul></li>
<li>Can be used to model airborne and directly transmitted diseases, for example, measles.</li>
</ul>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>We will only use the density-dependent formulation in this course.</p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<p><a href="images/model_diagrams/model_diagrams.003.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-9"><img data-src="images/model_diagrams/model_diagrams.003.jpeg"></a></p>
</section>
<section class="slide level2">

<h4 id="process-2-recovery">Process 2: Recovery</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/model_diagrams/model_diagrams.005.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-10"><img data-src="images/model_diagrams/model_diagrams.005.jpeg" class="quarto-figure quarto-figure-center" style="width:60.0%"></a></p>
</figure>
</div>
<p><span style="color:tomato;">Recovery</span>, governed by the recovery rate, <span class="math inline">\(\gamma\)</span> (rate at which infected individuals recover and become immune).</p>
</section>
<section class="slide level2">

<h5 id="some-notes-on-the-recovery-process">Some notes on the recovery process</h5>
<ul>
<li><p>If the duration of infection is <span class="math inline">\(\dfrac{1}{\gamma}\)</span>, then the rate at which infected individuals recover is <span class="math inline">\(\gamma\)</span>.</p></li>
<li><p>The average infectious period is often <span style="color:tomato;">estimated from epidemiological data</span>.</p></li>
<li><p>You will learn about parameter estimation in the model calibration lecture.</p></li>
</ul>
</section>
<section class="slide level2">

<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/epi_parameters.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Source: @anderson1982directly"><img data-src="images/epi_parameters.png" alt="Source: Anderson and May (1982)"></a></p>
<figcaption>Source: <span class="citation" data-cites="anderson1982directly">Anderson and May (<a href="#/references" role="doc-biblioref" onclick="">1982</a>)</span></figcaption>
</figure>
</div>
</section>
<section class="slide level2">

<h3 id="putting-it-all-together">Putting it all together</h3>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/model_diagrams/model_diagrams.006.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="An SIR model with transmission rate, \beta, and recovery rate, \gamma."><img data-src="images/model_diagrams/model_diagrams.006.jpeg" style="width:80.0%" alt="An SIR model with transmission rate, \beta, and recovery rate, \gamma."></a></p>
<figcaption>An SIR model with transmission rate, <span class="math inline">\(\beta\)</span>, and recovery rate, <span class="math inline">\(\gamma\)</span>.</figcaption>
</figure>
</div>
</section>
<section class="slide level2">

<h3 id="formulating-the-model-equations">Formulating the model equations</h3>
<p><span style="color:tomato;">Continuous time compartmental</span> models are formulated using <span style="color:tomato;">differential equations</span> that describe the change in the number of individuals in each compartment over time.</p>
</section>
<section class="slide level2">

<p>The SIR model can be formulated as:</p>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/model_diagrams/model_diagrams.006.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-13"><img data-src="images/model_diagrams/model_diagrams.006.jpeg" class="quarto-figure quarto-figure-center"></a></p>
</figure>
</div>
</div><div class="column" style="width:50%;">
<p><span class="math display">\[\begin{align}
\frac{dS}{dt} &amp; = \dot{S}  = \color{orange}{-\beta S I} \\
\frac{dI}{dt} &amp; = \dot{I}  = \color{orange}{\beta S I} - \color{blue}{\gamma I} \\
\frac{dR}{dt} &amp; = \dot{R} = \color{blue}{\gamma I}
\end{align}\]</span></p>
</div>
</div>
<p>where <span style="color:blue;"><span class="math inline">\(\beta\)</span></span> is the transmission rate, and <span style="color:blue;"><span class="math inline">\(\gamma\)</span></span> is the recovery rate.</p>
<aside class="notes">
<ul>
<li>We may or may not use the dot notation to denote the rate of change.</li>
<li>The terms are just <strong>inflows</strong> and <strong>outflows</strong>.</li>
<li><strong>Pause</strong> for questions and clarifications.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<p>The initial conditions are given by:</p>
<p><span class="math display">\[\begin{equation*}
\begin{aligned}
S(0) &amp; = N - 1,\\
I(0) &amp; = 1, \text{and} \\
R(0) &amp; = 0.
\end{aligned}
\end{equation*}\]</span></p>
<p>where <span class="math inline">\(N\)</span> is the total population size.</p>
</section>
<section class="slide level2">

<div class="callout callout-note no-icon callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<ul>
<li>We represent the compartments as <span style="color:tomato;">population sizes</span>.:
<ul>
<li>Some modellers often use <span style="color:tomato;">proportions</span> instead of population sizes as a way to remove the dimensions from the equations.</li>
</ul></li>
</ul>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="model-assumptions">Model assumptions</h3>
<ul>
<li>The population is closed: no births, deaths, or migration.
<ul>
<li>Implicitly: the epidemic occurs much faster than the time scale of births, deaths, or migration.</li>
</ul></li>
<li>Individuals are infectious immediately after infection and remain infectious until they recover.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>Mixing is <em>homogeneous</em>, i.e., <span style="color:tomato;">individuals mix randomly</span>:</p></li>
<li><p>Individuals have an equal probability of coming into contact with any other individual in the population.</p></li>
<li><p>Transition rates are constant and do not change over time.</p></li>
<li><p>Individuals acquire “lifelong” immunity after recovery.</p></li>
</ul>
<aside class="notes">
<p>The <span class="math inline">\(R\)</span> compartment is an absorbing state in this model.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<section id="discussion-2" class="callout-callout">
<h4>Discussion</h4>
<ul>
<li>What diseases do you think the SIR model is appropriate for?</li>
</ul>
</section>
</section>
<section class="slide level2">

<ul>
<li>The SIR model is appropriate for diseases that confer immunity after recovery. For example, measles and chicken pox.</li>
<li>Popularised by Kermack and McKendrick in 1927 <span class="citation" data-cites="kermack1927contribution">(<a href="#/references" role="doc-biblioref" onclick="">Kermack and McKendrick 1927</a>)</span>
<ul>
<li>A must-read paper for budding infectious disease modellers.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<h3 id="what-questions-can-we-answer-with-the-sir-model">What questions can we answer with the SIR model?</h3>
<ul>
<li>The SIR model can be used to understand the dynamics of an epidemic:
<ul>
<li>How long will the epidemic last?</li>
<li>How many individuals will be infected (final epidemic size)?</li>
<li>When will the epidemic reach its peak?</li>
</ul></li>
</ul>
</section>
<section id="solving-the-sir-model" class="slide level2">
<h2>Solving the SIR model</h2>
<ul>
<li><p>Compartmental models cannot be solved analytically.</p></li>
<li><p>We often perform two types of analyses to understand the <span style="color:tomato;">long term dynamics</span> of the model:</p>
<ul>
<li>Qualitative: threshold phenomena and analysis of equilibria (disease-free and endemic).<br>
</li>
<li>Numerical simulations.</li>
</ul></li>
</ul>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<ul>
<li>For this introductory course, we will use focus on simulation.</li>
</ul>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="threshold-phenomena">Threshold phenomena</h3>
<ul>
<li>Here, we study the conditions under which an epidemic will <span style="color:tomato;">grow or die out</span> using the model equations.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>Consider the case where <span class="math inline">\(I(0) = 1\)</span> individual is introduced into a population of size <span class="math inline">\(N\)</span> at time <span class="math inline">\(t = 0\)</span>.</p></li>
<li><p>That means in a completely susceptible population, we have <span class="math inline">\(S(0) = N - 1\)</span> susceptible individuals.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>At time 0, the disease will not spread if the rate of change of infections is negative, that is <span style="color:blue;"><span class="math inline">\(\dfrac{dI}{dt} &lt; 0\)</span></span>.</p></li>
<li><p>Recall from the SIR model that <span style="color:blue;"><span class="math inline">\(\dfrac{dI}{dt} = \beta S I - \gamma I\)</span></span>.</p></li>
<li><p>Let’s solve this equation at <span class="math inline">\(t = 0\)</span> by setting <span class="math inline">\(I = 1\)</span>, assuming <span style="color:blue;"><span class="math inline">\(\dfrac{dI}{dt} &lt; 0\)</span></span>.</p></li>
</ul>
</section>
<section class="slide level2">

<p>At <span class="math inline">\(t = 0\)</span>, we have</p>
<p><span class="math display">\[\begin{equation*}
\frac{dI}{dt} = \beta S I - \gamma I &lt; 0 \end{equation*}\]</span></p>
<p>Factor out <span class="math inline">\(I\)</span>, and we get</p>
<p><span class="math display">\[\begin{equation*}
I (\beta S - \gamma) &lt; 0
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<ul>
<li><p>Since at <span class="math inline">\(t=0\)</span><span class="math inline">\(, I &gt; 0\)</span>, we have <span style="color:blue;"><span class="math inline">\(S &lt; \dfrac{\gamma}{\beta}\)</span></span>.</p></li>
<li><p><span style="color:blue;"><span class="math inline">\(\dfrac{\gamma}{\beta}\)</span></span> is the relative removal rate.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p><strong>Interpretation</strong>:</p>
<ul>
<li>At <span class="math inline">\(t = 0\)</span>, <span class="math inline">\(S\)</span> must be less than <span class="math inline">\(\dfrac{\gamma}{\beta}\)</span> for the epidemic to die out.</li>
<li>If the rate of removal/recovery is greater than the transmission rate, the epidemic will die out.</li>
<li>Any infection that cannot transmit to more than one host is going to die out.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<p>For the SIR model, the quantity <span style="color:blue"><span class="math inline">\(\dfrac{\beta}{\gamma}\)</span></span> is called the <span style="color:tomato">reproduction number, <span class="math inline">\(R0\)</span></span> (pronounced “R naught” or “R zero”).</p>
</section>
<section class="slide level2">

<h4 id="the-basic-reproduction-number-r0">The basic reproduction number, R0</h4>
<ul>
<li><p>The basic reproduction number, <span style="color:blue"><span class="math inline">\(R0\)</span></span>, is the average number of secondary infections generated by a <span style="color:tomato">single primary infection</span> in a <span style="color:tomato">completely susceptible</span> population.</p></li>
<li><p>The basic reproduction number is a key quantity in infectious disease epidemiology.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>It is often represented as a single number or a range of high-low values.
<ul>
<li>For example, the <span class="math inline">\(R0\)</span> for measles is popularly known to be <span class="math inline">\(12\)</span> - <span class="math inline">\(18\)</span> <span class="citation" data-cites="guerra2017measlesR0">(<a href="#/references" role="doc-biblioref" onclick="">Guerra et al. 2017</a>)</span>.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><span class="math inline">\(R0\)</span> is often used to express the threshold phenomena in infectious disease epidemiology:
<ul>
<li>If <span class="math inline">\(R0 &gt; 1\)</span>, the epidemic will grow.</li>
<li>If <span class="math inline">\(R0 &lt; 1\)</span>, the epidemic will decline.</li>
</ul></li>
<li>A pathogen’s <span class="math inline">\(R0\)</span> value is determined by biological characteristics of the pathogen and the host’s behaviour.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Conceptually, <span class="math inline">\(R0\)</span> is given by</li>
</ul>
<p><span class="math display">\[
R0 \propto \dfrac{\text{Infection}}{\text{Contact}} \times \dfrac{\text{Contact}}{\text{Time}} \times \dfrac{\text{Time}}{\text{Infection}}
\]</span></p>
<ul>
<li><span class="math inline">\(R0\)</span> is unitless and dimensionless.</li>
</ul>
</section>
<section class="slide level2">

<h3 id="numerical-simulations">Numerical simulations</h3>
<ul>
<li><p>Numerical simulations can be performed with any programming language.</p></li>
<li><p>This course focuses on the R programming language because:</p></li>
<li><p>It is a popular language for data analysis and statistical computing.</p></li>
<li><p>It has a rich ecosystem of packages for solving differential equations.</p></li>
<li><p>It is free and open-source.</p></li>
<li><p>In R, we can use the <code>{deSolve}</code> package to solve the differential equations.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>To solve the model in R with <code>{deSolve}</code>, we will always need to define at least three things:
<ul>
<li>The model equations.</li>
<li>The initial parameter values.</li>
<li>The initial conditions (population sizes).</li>
</ul></li>
<li>Let’s do a code walk though in R using the script <code>01_sir.Rmd</code>.</li>
</ul>
</section>
<section class="slide level2">

<div class="callout callout-caution no-icon callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<p><strong>Discussion</strong></p>
</div>
<div class="callout-content">
<ul>
<li>What happens if we increase or decrease the value of <span class="math inline">\(R0\)</span>?</li>
<li>What happens if we increase or decrease the value of the infectious period?</li>
<li>How can we flatten the curve?</li>
</ul>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>We can flatten the curve by reducing the value of <span class="math inline">\(R0\)</span>.</li>
<li>In reality, we can reduce <span class="math inline">\(R0\)</span> by implementing control measures such as social distancing, wearing masks, and vaccination.</li>
<li>These control measures reduce the number of contacts between individuals or the probability of infection, which in turn reduces the transmission rate.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="the-susceptible-exposed-infected-recovered-seir-model" class="slide level2">
<h2>The Susceptible-Exposed-Infected-Recovered (SEIR) Model</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/infection_timeline.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14" title="Timeline of infection. Source: Keeling &amp; Rohani, 2008"><img data-src="images/infection_timeline.png" alt="Timeline of infection. Source: Keeling &amp; Rohani, 2008"></a></p>
<figcaption>Timeline of infection. Source: Keeling &amp; Rohani, 2008</figcaption>
</figure>
</div>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:30%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/corona_virus.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-15" title="The SARS-COV-2 virus"><img data-src="images/corona_virus.jpeg" alt="The SARS-COV-2 virus"></a></p>
<figcaption>The SARS-COV-2 virus</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><a href="images/ebola_virus.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-16" title="The Ebola virus"><img data-src="images/ebola_virus.jpeg" alt="The Ebola virus"></a></p>
<figcaption>The Ebola virus</figcaption>
</figure>
</div>
</div><div class="column" style="width:70%;">
<ul>
<li>Some diseases have an <span style="color:tomato;">latent/exposed period</span> during which individuals are <span style="color:tomato;">infected but not yet infectious</span>. Examples include pertussis, COVID-19, and Ebola.</li>
<li>Disease transmission does not occur during the latent period because of low levels of the virus in the host.</li>
</ul>
</div>
</div>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:50%;">
<p><a href="images/model_diagrams/model_diagrams.007.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-17"><img data-src="images/model_diagrams/model_diagrams.007.jpeg"></a></p>
</div><div class="column" style="width:50%;">
<ul>
<li>The SEIR model extends the SIR model to include an exposed compartment, <span style="color:blue;"><span class="math inline">\(E\)</span></span>.</li>
<li><span style="color:blue;"><span class="math inline">\(E\)</span></span>: infected but are not yet infectious.</li>
<li>Individuals stay in <span style="color:blue;"><span class="math inline">\(E\)</span></span> for <span style="color:blue;"><span class="math inline">\(1/\sigma\)</span></span> days before moving to <span class="math inline">\(I\)</span>.</li>
</ul>
</div>
</div>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:60%;">
<p><a href="images/model_diagrams/model_diagrams.008.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-18"><img data-src="images/model_diagrams/model_diagrams.008.jpeg" style="width:80.0%"></a></p>
</div><div class="column" style="width:40%;">
<p>Model equations: <span class="math display">\[\begin{align}
\frac{dS}{dt} &amp; = -\beta S I \\
\frac{dE}{dt} &amp; = \beta S I - \color{orange}{\sigma E} \\
\frac{dI}{dt} &amp; = \color{orange}{\sigma E} - \gamma I \\
\frac{dR}{dt} &amp; = \gamma I
\end{align}\]</span></p>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="seir-model-with-births-and-deaths">SEIR model with births and deaths</h3>
<ul>
<li><p>Let’s relax the assumption about births and deaths in the population.</p></li>
<li><p>We will assume that the susceptible population is replenished with new individuals at a constant rate, <span class="math inline">\(\mu\)</span>.</p></li>
<li><p>We will also assume that everyone dies at a constant rate, <span class="math inline">\(\mu\)</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<p>Our model schematic now looks like this:</p>
<p><a href="images/model_diagrams/model_diagrams.009.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-19"><img data-src="images/model_diagrams/model_diagrams.009.jpeg"></a></p>
<aside class="notes">
<p>Explain in terms of inflows and outflows</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:50%;">
<p><a href="images/model_diagrams/model_diagrams.009.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-20"><img data-src="images/model_diagrams/model_diagrams.009.jpeg"></a></p>
</div><div class="column" style="width:50%;">
<p>The model equations now become:</p>
<p><span class="math display">\[\begin{align}
\frac{dS}{dt} &amp; = \color{green}{\mu N} - \beta S I - \color{blue}{\mu} S \\
\frac{dE}{dt} &amp; = \beta S I - \sigma E - \color{blue}{\mu} E \\
\frac{dI}{dt} &amp; = \sigma E - \gamma I - \color{blue}{\mu} I \\
\frac{dR}{dt} &amp; = \gamma I - \color{blue}{\mu} R
\end{align}\]</span></p>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li><p>Let’s implement the SEIR model in R using the script <code>02_seir.Rmd</code>.</p></li>
<li><p>What is the <span class="math inline">\(R0\)</span> of this SEIR model?</p></li>
</ul>
</section>
<section class="slide level2">

<h3 id="the-r0-of-the-seir-model">The <span class="math inline">\(R0\)</span> of the SEIR Model</h3>
<ul>
<li><p>The method used to derive <span class="math inline">\(R0\)</span> in the SIR model is possible because it is easy to factor out <span class="math inline">\(I\)</span> from the equations.</p></li>
<li><p>In more complex models, this approach becomes impossible to use.</p></li>
<li><p>A more general framework is the next-generation matrix approach <span class="citation" data-cites="diekmann1990definition diekmann2010construction">(<a href="#/references" role="doc-biblioref" onclick="">Diekmann, Heesterbeek, and Metz 1990</a>; <a href="#/references" role="doc-biblioref" onclick="">Diekmann, Heesterbeek, and Roberts 2010</a>)</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<h4 id="the-next-generation-matrix-approach">The next-generation matrix approach</h4>
<ul>
<li><p>The next-generation matrix (NGM) approach focuses on the infected compartments.</p></li>
<li><p>It views the transmission process as a process where new infections are “born” from existing infections through generations.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Here are the steps to derive <span class="math inline">\(R0\)</span> using the NGM approach <span class="citation" data-cites="diekmann2010construction">(<a href="#/references" role="doc-biblioref" onclick="">Diekmann, Heesterbeek, and Roberts 2010</a>)</span>:
<ul>
<li>Identify the compartments that produce new infections and describe how infections are changing over time. This subset of the system is called the “infected subsystem”.</li>
<li>Linearise the infected subsystem around the disease-free equilibrium by generating the Jacobian matrix (See <a href="#/sec-jacobian" class="quarto-xref">Section&nbsp;3.4.2.1.1</a>).</li>
</ul></li>
</ul>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>The disease-free equilibrium is the state where no infections are present. For an SEIR model, this is denoted as <span class="math inline">\((S, E, I, R) = (N, 0, 0, 0)\)</span>, where <span class="math inline">\(N\)</span> is the total population size.</p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h5 id="sec-jacobian">The Jacobian matrix</h5>
<ul>
<li>The Jacobian matrix is a matrix of first-order partial derivatives of a function.</li>
</ul>
<p><span class="math display">\[\begin{equation*}
J = \begin{bmatrix}
\dfrac{\partial f_1}{\partial x_1} &amp; \dfrac{\partial f_1}{\partial x_2} &amp; \cdots &amp; \dfrac{\partial f_1}{\partial x_n} \\
\dfrac{\partial f_2}{\partial x_1} &amp; \dfrac{\partial f_2}{\partial x_2} &amp; \cdots &amp; \dfrac{\partial f_2}{\partial x_n} \\
\vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
\dfrac{\partial f_n}{\partial x_1} &amp; \dfrac{\partial f_n}{\partial x_2} &amp; \cdots &amp; \dfrac{\partial f_n}{\partial x_n}
\end{bmatrix}
\end{equation*}\]</span></p>
<p>where <span class="math inline">\(f_i\)</span> is the <span class="math inline">\(i\)</span>-th function of the function and <span class="math inline">\(x_i\)</span> is the <span class="math inline">\(i\)</span>-th variable.</p>
</section>
<section class="slide level2">

<ul>
<li><p>The linearised infected subsystem is then decomposed into two matrices <span class="math inline">\(\mathcal{F}\)</span> and <span class="math inline">\(\mathcal{V}\)</span>:</p>
<ul>
<li><span class="math inline">\(\mathcal{F}\)</span> represents the transmission matrix, i.e., the rate of appearance of new infections in a compartment.</li>
<li><span class="math inline">\(\mathcal{V}\)</span> is represents transitions, i.e., how infections move into other states.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>Next, solve <span class="math inline">\(\mathcal{F}\mathcal{V}^{-1}\)</span> to obtain the next-generation matrix, <span class="math inline">\(\mathcal{G}\)</span>.</p></li>
<li><p>Finally, determine the spectral radius, <span class="math inline">\(\rho\)</span>, of <span class="math inline">\(\mathcal{G}\)</span>, which is the <span style="color:tomato;">dominant (largest absolute) eigenvalue</span> of the NGM.</p></li>
<li><p>The dominant eigenvalue of the NGM is the basic reproduction number, <span class="math inline">\(R0\)</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<h5 id="steps-to-determine-the-spectral-radius">Steps to determine the Spectral Radius</h5>
<ul>
<li>Find the Eigenvalues:
<ul>
<li>Given a square matrix <span class="math inline">\(A\)</span>, compute the eigenvalues <span class="math inline">\(\lambda\)</span> by solving the characteristic equation:</li>
</ul></li>
</ul>
<p><span class="math display">\[\begin{equation*}
\begin{aligned}
\text{det}(A - \lambda I) = 0
\end{aligned}
\end{equation*}\]</span></p>
<p>where <span class="math inline">\(I\)</span> is the identity matrix of the same dimension as <span class="math inline">\(A\)</span>, and <span class="math inline">\(\text{det}\)</span> denotes the determinant.</p>
</section>
<section class="slide level2">

<h5 id="steps-to-determine-the-spectral-radius-contd">Steps to determine the Spectral Radius (cont’d)</h5>
<ul>
<li>Compute the Spectral Radius, <span class="math inline">\(\rho(A) = \max \{ |\lambda_1|, |\lambda_2|, \ldots, |\lambda_n|\)</span></li>
</ul>
<p>where <span class="math inline">\(\lambda_1, \lambda_2, \ldots, \lambda_n\)</span> are the eigenvalues of <span class="math inline">\(A\)</span>, and <span class="math inline">\(|\lambda_i|\)</span> denotes the absolute value of the eigenvalue <span class="math inline">\(\lambda_i\)</span>.</p>
<p>Let’s apply these ideas to derive the <span class="math inline">\(R0\)</span> of the SEIR model.</p>
</section>
<section class="slide level2">

<h4 id="deriving-the-r0-of-the-seir-model">Deriving the R0 of the SEIR model</h4>
<p>Step 1: identify the infected compartments <span class="math inline">\(E\)</span> and <span class="math inline">\(I\)</span>:</p>
<p><span class="math display">\[\begin{equation*}
\begin{aligned}
\frac{dE}{dt} &amp;= \beta S I - \sigma E, \\
\frac{dI}{dt} &amp;= \sigma E - \gamma I.
\end{aligned}
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<p>Step 2: Decompose the system into <span class="math inline">\(\mathcal{F}\)</span>, and <span class="math inline">\(\mathcal{V}\)</span>:</p>
<p>Define:</p>
<ul>
<li><span class="math inline">\(\mathcal{F}\)</span>: new infection terms</li>
<li><span class="math inline">\(\mathcal{V}\)</span>: transition terms</li>
</ul>
<p><span class="math display">\[\begin{equation*}
\mathcal{F} = \begin{pmatrix}
\beta S I \\
0
\end{pmatrix},
\quad
\mathcal{V} = \begin{pmatrix}
\sigma E \\
\sigma E - \gamma I
\end{pmatrix}
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<p>Evaluate at the disease-free equilibrium (DFE) <span class="math inline">\((S, E, I, R) = (S_0, 0, 0, 0)\)</span>:</p>
<p><span class="math display">\[\begin{equation*}
\mathcal{F} = \begin{pmatrix}
0 &amp; \beta S_0 \\
0 &amp; 0
\end{pmatrix},
\quad
\mathcal{V} = \begin{pmatrix}
\sigma &amp; 0 \\
-\sigma &amp; \gamma
\end{pmatrix}
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<p>We aim to solve</p>
<p><span class="math display">\[\begin{equation*}
\mathcal{G} = \mathcal{F} \mathcal{V}^{-1}
\end{equation*}\]</span></p>
<p>Let’s compute <span class="math inline">\(\mathcal{V}^{-1}\)</span>:</p>
<p><span class="math display">\[\begin{equation*}
\mathcal{V}^{-1} = \begin{pmatrix}
\frac{1}{\sigma} &amp; 0 \\
\frac{1}{\gamma} &amp; \frac{1}{\sigma}
\end{pmatrix}
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<p><span class="math display">\[\begin{equation*}
\mathcal{G} = \begin{pmatrix}
0 &amp; \beta S_0 \\
0 &amp; 0
\end{pmatrix}
\begin{pmatrix}
\frac{1}{\sigma} &amp; 0 \\
\frac{1}{\gamma} &amp; \frac{1}{\sigma}
\end{pmatrix}
= \begin{pmatrix}
\frac{\beta S_0}{\gamma} &amp; \frac{\beta S_0}{\sigma} \\
0 &amp; 0
\end{pmatrix}
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<p><span class="math inline">\(R_0\)</span> is the spectral radius (dominant eigenvalue) of <span class="math inline">\(\mathcal{G}\)</span>:</p>
<p><span class="math display">\[\begin{equation*}
R_0 = \rho(\mathcal{G}) = \frac{\beta S_0}{\gamma}
\end{equation*}\]</span></p>
</section>
<section class="slide level2">

<h3 id="numerical-simulations-1">Numerical simulations</h3>
<ul>
<li><p>We can use the same approach as the SIR model to simulate the SEIR model.</p></li>
<li><p>Let’s do a code walk though in R.</p></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="controlling-epidemics" class="title-slide slide level1 center">
<h1>Controlling epidemics</h1>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<ul>
<li>For a recap on the various control measures, refer to <a href="#/sec-control-measures" class="quarto-xref">Section&nbsp;1.2</a>.</li>
</ul>
</div>
</div>
</div>
</section>
<section id="vaccination" class="slide level2">
<h2>Vaccination</h2>
<div class="{callout-note}">
<ul>
<li>For a background on vaccination, refer to <a href="#/sec-vaccination" class="quarto-xref">Section&nbsp;1.2.1.1</a>.</li>
</ul>
</div>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:40%;">
<p><a href="images/vaccine.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-21"><img data-src="images/vaccine.jpeg" style="width:150.0%"></a></p>
</div><div class="column" style="width:60%;">
<ul>
<li>Vaccination is one of the most effective ways to control infectious diseases.</li>
<li>Conceptually, vaccination works to reduce the number of susceptible individuals, <span class="math inline">\(S\)</span>.</li>
</ul>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li><p>There are different types of vaccination strategies, including:</p>
<ul>
<li><span style="color:tomato">pediatric</span> vaccination: vaccinating children to prevent the spread of diseases.</li>
<li><span style="color:tomato">mass/random</span> vaccination: vaccinating a large proportion of the population.</li>
<li><span style="color:tomato">targeted</span> vaccination: vaccinating specific groups of individuals, example, healthcare workers.</li>
<li><span style="color:tomato">pulse</span> vaccination: periodically vaccinating a large number of individuals.</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>Let’s consider the case of mass/random vaccination.</p></li>
<li><p>Compartmental models can be extended to capture this by adding a new compartment, <span class="math inline">\(V\)</span>.</p></li>
<li><p>Let’s consider the SEIR model with vaccination.</p></li>
</ul>
</section>
<section class="slide level2">

<h3 id="the-susceptible-exposed-infected-recovered-vaccinated-seirv-model">The Susceptible-Exposed-Infected-Recovered-Vaccinated (SEIRV) Model</h3>
<div class="columns">
<div class="column" style="width:40%;">
<p><a href="images/model_diagrams/model_diagrams.003.png" class="lightbox" data-gallery="quarto-lightbox-gallery-22"><img data-src="images/model_diagrams/model_diagrams.003.png"></a></p>
</div><div class="column" style="width:60%;">
<ul>
<li><p>The SEIRV model is simply the SEIR model with a vaccinated compartment, <span class="math inline">\(V\)</span>.</p></li>
<li><p>The vaccinated compartment represents previously susceptible individuals who have been vaccinated and are immune to the disease.</p></li>
<li><p>The vaccinated compartment is not infectious and does not move to the exposed or infectious compartments.</p></li>
<li><p>The vaccinated compartment is replenished by the rate of vaccination, <span class="math inline">\(\eta\)</span>.</p></li>
</ul>
</div>
</div>
</section>
<section class="slide level2">

<p>The model diagram and equations are as follows:</p>
<div class="columns">
<div class="column" style="width:40%;">
<p><span class="math display">\[\begin{align}
\frac{dS}{dt} &amp; = -\beta \frac{S I}{N} - \eta S \\
\frac{dE}{dt} &amp; = \beta \frac{S I}{N} - \sigma E \\
\frac{dI}{dt} &amp; = \sigma E - \gamma I \\
\frac{dR}{dt} &amp; = \gamma I \\
\frac{dV}{dt} &amp; = \eta S
\end{align}\]</span></p>
</div><div class="column" style="width:60%;">
<p><a href="images/model_diagrams/model_diagrams.003.png" class="lightbox" data-gallery="quarto-lightbox-gallery-23"><img data-src="images/model_diagrams/model_diagrams.003.png"></a></p>
</div>
</div>
<p>where <span class="math inline">\(\eta\)</span> is the rate of vaccination.</p>
</section>
<section class="slide level2">

<div class="callout callout-caution no-icon callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<p><strong>Discussion</strong></p>
</div>
<div class="callout-content">
<ul>
<li><p>What are some of the assumptions of the SEIRV model?</p></li>
<li><p>What are the implications of these assumptions for the model’s predictions?</p></li>
</ul>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="numerical-simulations-2">Numerical simulations</h3>
<ul>
<li><p>We can use the same approach as the SIR and SEIR models to simulate the SEIRV model.</p></li>
<li><p>Let’s do a code walk though in R.</p></li>
</ul>
</section>
<section id="modelling-non-pharmaceutical-interventions-npis" class="slide level2">
<h2>Modelling Non-Pharmaceutical Interventions (NPIs)</h2>
<div class="{callout-note}">
<ul>
<li>For a background on non-pharmaceutical interventions, refer to <a href="#/sec-npi" class="quarto-xref">Section&nbsp;1.2.2</a>.</li>
</ul>
</div>
<ul>
<li><p>Conceptually, NPIs usually act to either reduce the transmission rate, <span class="math inline">\(\beta\)</span> or prevent infected individuals from transmitting.</p></li>
<li><p>NPIs like quarantine, social distancing and movement restrictions can reduce the transmission rate, <span class="math inline">\(\beta\)</span>, by reducing the contact rate between susceptible and infectious individuals.</p></li>
<li><p>Hygiene measures reduce the probability of transmission per contact, thereby reducing the transmission rate, <span class="math inline">\(\beta\)</span>, since <span class="math inline">\(\beta = c \times p\)</span>.</p></li>
</ul>
</section>
<section class="slide level2 scrollable">

<ul>
<li><p>Let’s consider two scenarios that will extend the SIR model to include NPIs:</p>
<ol type="1">
<li>Modifying the transmission rate, <span class="math inline">\(\beta\)</span>.</li>
<li>Preventing infected individuals from transmitting through quarantine.</li>
</ol></li>
</ul>
</section>
<section class="slide level2">

<h3 id="modifying-the-transmission-rate">Modifying the transmission rate</h3>
<ul>
<li><p>NPIs such as social distancing, mask-wearing, and hand hygiene can reduce the transmission rate, <span class="math inline">\(\beta\)</span>.</p></li>
<li><p>We can model this by making the transmission rate a function of time so that <span class="math inline">\(\beta(t)\)</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<p>The modified SIR model with a reduced transmission rate is as follows:</p>
<p><span class="math display">\[\begin{align}
\frac{dS}{dt} &amp; = -\beta(t) \frac{S I}{N} \\
\frac{dI}{dt} &amp; = \beta (t) \frac{S I}{N} - \gamma I \\
\frac{dR}{dt} &amp; = \gamma I
\end{align}\]</span></p>
</section>
<section class="slide level2">

<ul>
<li><p>The simplest form is to reduce <span class="math inline">\(\beta\)</span> by an NPI efficacy, say <span class="math inline">\(\epsilon\)</span>.</p></li>
<li><p>Assuming the NPI is implemented between <span class="math inline">\(t_{npi_start}\)</span> and <span class="math inline">\(t_{npi_end}\)</span>, it means that <span class="math inline">\(\beta\)</span> remains the same before that period and is reduced to <span class="math inline">\(( 1- \epsilon)\)</span> during the period of the NPI.</p></li>
<li><p>With this knowledge, we can define <span class="math inline">\(\beta(t)\)</span> mathematically as:</p></li>
</ul>
<p><span class="math display">\[\begin{align*}
\beta(t) = \begin{cases}
\beta &amp; \text{if } t &lt; t_{\test{npi_start}} | t &gt; t_{\test{npi_end}} \\
(1 - \epsilon) \beta &amp; \text{if } t_{\test{npi_start}} \le t \le t_{\test{npi_end}}
\end{cases}
\end{align*}\]</span></p>
</section>
<section class="slide level2">

<ul>
<li><p>Let’s translate this to R code.</p></li>
<li><p>We will build on our previous SIR model code and modify it to include the time-varying transmission rate.</p></li>
</ul>
</section>
<section class="slide level2">

<p>We will start with the main model function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1" data-code-line-numbers="1-4,6-12|5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href=""></a><span class="co"># Define the SIR model differential equations</span></span>
<span id="cb1-2"><a href=""></a>sir_with_npi <span class="ot">&lt;-</span> <span class="cf">function</span>(time, state, parameters) {</span>
<span id="cb1-3"><a href=""></a>  <span class="fu">with</span>(<span class="fu">as.list</span>(<span class="fu">c</span>(state, parameters)), {</span>
<span id="cb1-4"><a href=""></a>    <span class="co"># Time-varying transmission rate</span></span>
<span id="cb1-5"><a href=""></a>    beta <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(time <span class="sc">&lt;</span> t_NPI_start <span class="sc">|</span> time <span class="sc">&gt;</span> t_NPI_end, beta_0, (<span class="dv">1</span> <span class="sc">-</span> epsilon) <span class="sc">*</span> beta_0)</span>
<span id="cb1-6"><a href=""></a>    <span class="co"># SIR model</span></span>
<span id="cb1-7"><a href=""></a>    dS <span class="ot">&lt;-</span> <span class="sc">-</span>beta <span class="sc">*</span> S <span class="sc">*</span> I <span class="sc">/</span> N</span>
<span id="cb1-8"><a href=""></a>    dI <span class="ot">&lt;-</span> beta <span class="sc">*</span> S <span class="sc">*</span> I <span class="sc">/</span> N <span class="sc">-</span> gamma <span class="sc">*</span> I</span>
<span id="cb1-9"><a href=""></a>    dR <span class="ot">&lt;-</span> gamma <span class="sc">*</span> I</span>
<span id="cb1-10"><a href=""></a>    <span class="fu">list</span>(<span class="fu">c</span>(dS, dI, dR))</span>
<span id="cb1-11"><a href=""></a>  })</span>
<span id="cb1-12"><a href=""></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section class="slide level2">

<p>Next, we will define the parameters and initial conditions. Note that we will also define the time at which the NPI is implemented, <code>t_NPI_start</code> and <code>t_NPI_end</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href=""></a><span class="co"># The initial conditions</span></span>
<span id="cb2-2"><a href=""></a>N  <span class="ot">&lt;-</span> <span class="dv">1000</span>  <span class="co"># Population size</span></span>
<span id="cb2-3"><a href=""></a><span class="co"># Initial conditions for S, I, R</span></span>
<span id="cb2-4"><a href=""></a>inits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">S =</span> N <span class="sc">-</span> <span class="dv">1</span>, <span class="at">I =</span> <span class="dv">1</span>, <span class="at">R =</span> <span class="dv">0</span>)</span>
<span id="cb2-5"><a href=""></a></span>
<span id="cb2-6"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb2-7"><a href=""></a>R0 <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb2-8"><a href=""></a>infectious_period <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb2-9"><a href=""></a><span class="co"># Remember gamma = 1/ infectious_period as discussed earlier</span></span>
<span id="cb2-10"><a href=""></a>gamma <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>infectious_period </span>
<span id="cb2-11"><a href=""></a></span>
<span id="cb2-12"><a href=""></a><span class="co"># We will use R0 = beta N / gamma  instead because it is easier to interpret.</span></span>
<span id="cb2-13"><a href=""></a><span class="co"># beta is not directly interpretable because it is a composite of a number of factors.</span></span>
<span id="cb2-14"><a href=""></a>params <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-15"><a href=""></a>  <span class="at">beta_0 =</span> R0 <span class="sc">*</span> gamma <span class="sc">/</span> N,</span>
<span id="cb2-16"><a href=""></a>  <span class="at">gamma =</span> gamma,</span>
<span id="cb2-17"><a href=""></a>  <span class="at">epsilon =</span> <span class="fl">5E-4</span></span>
<span id="cb2-18"><a href=""></a>)</span>
<span id="cb2-19"><a href=""></a></span>
<span id="cb2-20"><a href=""></a><span class="co"># Time steps to return results</span></span>
<span id="cb2-21"><a href=""></a>dt <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">150</span></span>
<span id="cb2-22"><a href=""></a></span>
<span id="cb2-23"><a href=""></a><span class="co"># Time at which the NPI is implemented</span></span>
<span id="cb2-24"><a href=""></a>npi_times <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">t_NPI_start =</span> <span class="dv">30</span>, <span class="at">t_NPI_end =</span> <span class="dv">51</span>) <span class="co"># 21 days of NPI  </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section class="slide level2">

<p>Now, let’s solve the model equations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href=""></a><span class="co"># Load necessary packages</span></span>
<span id="cb3-2"><a href=""></a><span class="fu">library</span>(deSolve)</span>
<span id="cb3-3"><a href=""></a><span class="co"># Integrate the SIR equations over the time grid</span></span>
<span id="cb3-4"><a href=""></a>output <span class="ot">&lt;-</span> <span class="fu">lsoda</span>(</span>
<span id="cb3-5"><a href=""></a>  <span class="at">y =</span> inits,</span>
<span id="cb3-6"><a href=""></a>  <span class="at">times =</span> dt,</span>
<span id="cb3-7"><a href=""></a>  <span class="at">func =</span> sir_with_npi,</span>
<span id="cb3-8"><a href=""></a>  <span class="at">parms =</span> <span class="fu">c</span>(params, npi_times)</span>
<span id="cb3-9"><a href=""></a>)</span>
<span id="cb3-10"><a href=""></a></span>
<span id="cb3-11"><a href=""></a><span class="co"># Convert the output to a data frame</span></span>
<span id="cb3-12"><a href=""></a>output_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(output)</span>
<span id="cb3-13"><a href=""></a><span class="fu">head</span>(output_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  time        S         I         R
1    1 999.0000 1.0000000 0.0000000
2    2 998.9973 0.8693567 0.1333068
3    3 998.9950 0.7557805 0.2491984
4    4 998.9930 0.6570417 0.3499502
5    5 998.9913 0.5712029 0.4375391
6    6 998.9897 0.4965784 0.5136850</code></pre>
</div>
</div>
</section>
<section class="slide level2">

<p>What do you observe in the output?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href=""></a><span class="co"># Load necessary packages</span></span>
<span id="cb5-2"><a href=""></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb5-3"><a href=""></a><span class="co"># Convert the output to long format</span></span>
<span id="cb5-4"><a href=""></a>output_df <span class="ot">&lt;-</span> output_df <span class="sc">|&gt;</span> <span class="fu">pivot_longer</span>(</span>
<span id="cb5-5"><a href=""></a>  <span class="at">cols =</span> <span class="sc">-</span>time,</span>
<span id="cb5-6"><a href=""></a>  <span class="at">names_to =</span> <span class="st">"compartment"</span>,</span>
<span id="cb5-7"><a href=""></a>  <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb5-8"><a href=""></a>)</span>
<span id="cb5-9"><a href=""></a><span class="co"># Plot the data using ggplot2</span></span>
<span id="cb5-10"><a href=""></a>sir_npi_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb5-11"><a href=""></a>  <span class="at">data =</span> output_df,</span>
<span id="cb5-12"><a href=""></a>  <span class="fu">aes</span>(</span>
<span id="cb5-13"><a href=""></a>    <span class="at">x =</span> time,</span>
<span id="cb5-14"><a href=""></a>    <span class="at">y =</span> value,</span>
<span id="cb5-15"><a href=""></a>    <span class="at">color =</span> compartment</span>
<span id="cb5-16"><a href=""></a>  )) <span class="sc">+</span></span>
<span id="cb5-17"><a href=""></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb5-18"><a href=""></a>  <span class="co"># Add a vertical line to indicate the times of NPI implementation</span></span>
<span id="cb5-19"><a href=""></a>  <span class="fu">geom_vline</span>(</span>
<span id="cb5-20"><a href=""></a>    <span class="at">xintercept =</span> npi_times[<span class="st">"t_NPI_start"</span>],</span>
<span id="cb5-21"><a href=""></a>    <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb5-22"><a href=""></a>    <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb5-23"><a href=""></a>  ) <span class="sc">+</span></span>
<span id="cb5-24"><a href=""></a>  <span class="fu">geom_vline</span>(</span>
<span id="cb5-25"><a href=""></a>    <span class="at">xintercept =</span> npi_times[<span class="st">"t_NPI_end"</span>],</span>
<span id="cb5-26"><a href=""></a>    <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb5-27"><a href=""></a>    <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb5-28"><a href=""></a>  ) <span class="sc">+</span> </span>
<span id="cb5-29"><a href=""></a>  <span class="co"># Add labels</span></span>
<span id="cb5-30"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb5-31"><a href=""></a>    <span class="at">title =</span> <span class="st">"SIR Model with NPI"</span>,</span>
<span id="cb5-32"><a href=""></a>    <span class="at">x =</span> <span class="st">"Time (days)"</span>,</span>
<span id="cb5-33"><a href=""></a>    <span class="at">y =</span> <span class="st">"Number of individuals"</span>,</span>
<span id="cb5-34"><a href=""></a>    <span class="at">color =</span> <span class="st">"Compartment"</span></span>
<span id="cb5-35"><a href=""></a>  ) <span class="sc">+</span></span>
<span id="cb5-36"><a href=""></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section class="slide level2">

<p>Let’s plot the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href=""></a><span class="fu">plot</span>(sir_npi_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><a href="slides_files/figure-revealjs/sir-npi-beta-plot-show-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-24"><img data-src="slides_files/figure-revealjs/sir-npi-beta-plot-show-1.png" width="960"></a></p>
</figure>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h4 id="scenarios-with-reduced-transmission-rate">Scenarios with reduced transmission rate</h4>
<ul>
<li><p>We can use the same approach as the SIR model to simulate the model with a reduced transmission rate.</p></li>
<li><p>Let’s do a code walk though in R.</p></li>
</ul>
</section>
<section class="slide level2">

<h3 id="npis-as-compartments">NPIs as compartments</h3>
<ul>
<li><p>In the previous example, we retained the SIR model structure and modified the transmission rate.</p></li>
<li><p>We can also model NPIs as compartments in the model. This is useful when we want to treat individuals affected by the NPIs differently.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>For example, quarantine is an NPI that prevents infected individuals from transmitting the disease.</p></li>
<li><p>This means that infected individuals under quarantine do not contribute to the transmission of the disease and need to be removed from the infected compartment.</p></li>
<li><p>Moving quarantined individuals to a separate compartment allows us to track them separately in the model and apply relevant parameters.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>We can model this by introducing a new compartment, <span class="math inline">\(Q\)</span>, for quarantining infected individuals.</p></li>
<li><p>Let’s assume, infected individuals move to the quarantine compartment at a rate, <span class="math inline">\(\delta\)</span>.</p></li>
<li><p>Infected individuals in the quarantine compartment do not transmit the disease.</p></li>
</ul>
</section>
<section class="slide level2">

<p>The modified SIR model with quarantine is as follows:</p>
<p><span class="math display">\[\begin{align*}
\frac{dS}{dt} &amp; = -\beta \frac{S I}{N} \\
\frac{dI}{dt} &amp; = \beta \frac{S I}{N} - \gamma I - \delta I \\
\frac{dR}{dt} &amp; = \gamma I + \tau Q \\
\frac{dQ}{dt} &amp; = \delta I - \tau Q
\end{align*}\]</span></p>
<p>where <span class="math inline">\(\delta\)</span> is the rate at which infected individuals move to the quarantine compartment, and <span class="math inline">\(\tau\)</span> is the rate at which individuals recover from quarantine.</p>
</section>
<section class="slide level2">

<h4 id="scenarios-with-quarantine">Scenarios with quarantine</h4>
<ul>
<li><p>We can use the same approach as the SIR model to simulate the model with quarantine.</p></li>
<li><p>Let’s do a code walk though in R.</p></li>
</ul>
</section></section>
<section>
<section id="an-overview-of-host-heterogeneity" class="title-slide slide level1 center">
<h1>An overview of host heterogeneity</h1>
<ul>
<li><p>The models we have discussed so far assume that all individuals in the population are identical.</p></li>
<li><p>However, in reality, individuals differ in their susceptibility to infection and their ability to transmit the disease.</p></li>
<li><p>It is essential to capture this heterogeneity in the models in order for the models to be more realistic and useful for decision-making.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>This can be captured by incorporating heterogeneity into the models.</p></li>
<li><p>Heterogeneity is often captured by stratifying the population into different groups.</p></li>
</ul>
</section>
<section id="age-structure" class="slide level2">
<h2>Age structure</h2>
<ul>
<li><p>For many infectious diseases, the risk of infection and the severity of the disease vary by age.</p></li>
<li><p>Hence, it is essential to capture age structure in the models.</p></li>
<li><p>To do this, we divide the population into different age groups and model the disease dynamics within each age group.</p></li>
<li><p>Let’s extend the SIR model to include age structure.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>We will divide the population into <span class="math inline">\(n\)</span> age groups.</p></li>
<li><p>Because the homogeneous model has 3 compartments, the age structured one will have <span class="math inline">\(3n\)</span> compartments: <span class="math inline">\(S_1, I_1, R_1, S_2, I_2, R_2, ..., S_n, I_n, R_n\)</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>The (compact) model equations are as follows:</li>
</ul>
<p><span class="math display">\[\begin{align}
\dfrac{dS_i}{dt} &amp;= -\sum_{j=1}^{n} \beta_{ji} S_i I_j \\
\dfrac{dI_i}{dt} &amp;= \sum_{j=1}^{n} \beta_{ji} S_i I_j - \gamma I_i \\
\dfrac{dR_i}{dt} &amp;= \gamma I_i
\end{align}\]</span></p>
</section>
<section class="slide level2">

<ul>
<li><p>The model can be used to study the impact of age structure on the dynamics of the epidemic.</p></li>
<li><p>For example, we can study the impact of vaccinating different age groups on the dynamics of the epidemic.</p></li>
</ul>
</section>
<section id="other-heterogeneities" class="slide level2">
<h2>Other Heterogeneities</h2>
<ul>
<li><p>Other forms of heterogeneity that can be incorporated into the models include:</p>
<ul>
<li>Spatial heterogeneity</li>
<li>Temporal heterogeneity</li>
<li>Contact heterogeneity</li>
</ul></li>
</ul>
</section></section>
<section>
<section id="wrap-up-and-qa" class="title-slide slide level1 center">
<h1>Wrap-Up and Q&amp;A</h1>

</section>
<section id="takeaways" class="slide level2">
<h2>Takeaways</h2>
<p>In the last two days, we have covered a lot of ground. Here are some key takeaways:</p>
<ul>
<li><p>Infectious diseases are a major public health concern that can have devastating consequences.</p></li>
<li><p>Mathematical models are essential tools for studying the dynamics of infectious diseases and informing public health decision-making.</p></li>
<li><p>Models are simplifications of reality that help us understand complex systems.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>We have discussed several compartmental models, including the SIR, SEIR, and SEIRV models.</p></li>
<li><p>The SIR model is a simple compartmental model that divides the population into three compartments: susceptible, infected, and recovered.</p></li>
<li><p>The SEIR model extends the SIR model by adding an exposed compartment.</p></li>
<li><p>We can model various pharmaceutical and non-pharmaceutical interventions (NPIs) by modifying the transmission rate or adding new compartments.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>We have discussed the basic reproduction number, <span class="math inline">\(R0\)</span>, which is a key parameter in infectious disease epidemiology.</p></li>
<li><p><span class="math inline">\(R0\)</span> is the average number of secondary infections produced by a single infected individual in a completely susceptible population.</p></li>
<li><p>If <span class="math inline">\(R0 &gt; 1\)</span>, the disease will spread in the population; if <span class="math inline">\(R0 &lt; 1\)</span>, the disease will die out.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>Deriving the basic reproduction number, <span class="math inline">\(R0\)</span>, is an essential step in understanding the dynamics of infectious diseases.</p></li>
<li><p>Deriving <span class="math inline">\(R0\)</span> for the simple SIR model is simple as we just need to study the threshold phenomena.</p></li>
<li><p>For more complex models, we can use the next-generation matrix approach to derive <span class="math inline">\(R0\)</span>.</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>Often, homogeneous models are not sufficient to capture the complexity of infectious diseases.</p></li>
<li><p>Incorporating heterogeneity into the models is essential for capturing the complexity of infectious diseases.</p></li>
<li><p>Age structure is a common form of heterogeneity that can be incorporated into the models.</p></li>
<li><p>Other forms of heterogeneity include spatial, temporal, and contact heterogeneity.</p></li>
</ul>
</section>
<section id="what-skills-are-needed-to-build-and-use-infectious-disease-models" class="slide level2">
<h2>What skills are needed to build and use infectious disease models?</h2>
<h3 id="mathematical-and-statistical-skills">Mathematical and statistical skills</h3>
<ul>
<li>Differential equations.</li>
<li>Probability and statistics.</li>
<li>Stochastic processes.</li>
<li>Numerical analysis</li>
<li>Time series analyses</li>
<li>Survival analysis</li>
</ul>
</section>
<section class="slide level2">

<h3 id="programming-skills">Programming skills</h3>
<ul>
<li>Proficiency in at least one programming language (e.g., R, Python, Julia, C++).</li>
<li>Experience with version control (e.g., Git).</li>
<li>Experience with data manipulation and visualization.</li>
</ul>
</section>
<section class="slide level2">

<h3 id="domain-knowledge-in-infectious-diseases">Domain knowledge in infectious diseases</h3>
<ul>
<li>Immunology</li>
<li>Epidemiology</li>
<li>Virology</li>
<li>Genomics</li>
</ul>
</section>
<section class="slide level2">

<h3 id="other-subject-areas">Other subject areas</h3>
<ul>
<li>Communication skills:
<ul>
<li>Ability to communicate complex ideas to non-experts.</li>
<li>Ability to write clear and concise reports.</li>
<li>Ability to present results to a diverse audience.</li>
</ul></li>
<li>Public health.</li>
<li>Health economics.</li>
<li>Policy analysis.</li>
</ul>
</section></section>
<section>
<section id="list-of-resources" class="title-slide slide level1 center">
<h1>List of Resources</h1>

</section>
<section id="textbooks" class="slide level2">
<h2>Textbooks</h2>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="2">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><a href="images/Anderson_and_May.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-25"><img data-src="images/Anderson_and_May.jpeg"></a></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><a href="images/ID_modelling_Vynnycky_and_White.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-26"><img data-src="images/ID_modelling_Vynnycky_and_White.jpeg"></a></p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="2">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><a href="images/Rohani_and_Keeling.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-27"><img data-src="images/Rohani_and_Keeling.jpeg"></a></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><a href="images/Epidemics_Ottar.jpg" class="lightbox" data-gallery="quarto-lightbox-gallery-28"><img data-src="images/Epidemics_Ottar.jpg"></a></p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li><p><a href="https://www.amazon.co.uk/Modeling-Infectious-Diseases-Humans-Animals/dp/0691116172">Modeling Infectious Diseases in Humans and Animals by Matt Keeling and Pejman Rohani</a></p></li>
<li><p><a href="https://link.springer.com/book/10.1007/978-3-031-12056-5">Epidemics: Models and Data Using R by Ottar N. Bjornstad</a></p></li>
<li><p><a href="https://www.amazon.co.uk/Introduction-Infectious-Disease-Modelling/dp/0198565763">Infectious Disease Modelling by Emilia Vynnycky and Richard White</a></p></li>
<li><p><a href="https://www.amazon.co.uk/Infectious-Diseases-Humans-Dynamics-Control/dp/019854040X">Infectious Diseases of Humans: Dynamics and Control by Roy M. Anderson and Robert M. May</a></p></li>
</ul>
</section>
<section id="papers-and-articles" class="slide level2 scrollable">
<h2>Papers and Articles</h2>
<h3 id="modelling-infectious-disease-transmission">Modelling infectious disease transmission</h3>
<ul>
<li><a href="https://www.frontiersin.org/articles/10.3389/fvets.2020.546651/full">Kirkeby, C., Brookes, V. J., Ward, M. P., Dürr, S., &amp; Halasa, T. (2021). A practical introduction to mechanistic modeling of disease transmission in veterinary science. Frontiers in veterinary science, 7, 546651.</a></li>
<li><a href="https://vtechworks.lib.vt.edu/items/61e9ca00-ef21-4356-bcd7-a9294a1d2f17">Blackwood, J. C., &amp; Childs, L. M. (2018). An introduction to compartmental modeling for the budding infectious disease modeler.</a></li>
<li><a href="https://doi.org/10.1038/nrmicro1845">Grassly, N. C., &amp; Fraser, C. (2008). Mathematical models of infectious disease transmission. Nature Reviews Microbiology, 6(6), 477–487.</a></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><a href="https://doi.org/10.1126/science.abb5659">Cobey, S. (2020). Modeling infectious disease dynamics. Science, 368(6492), 713–714.</a></li>
<li><a href="https://doi.org/10.1038/s41592-020-0822-z">Bjørnstad, O. N., Shea, K., Krzywinski, M., &amp; Altman, N. (2020). Modeling infectious epidemics. Nature Methods, 17(5), 455–456.</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1008539">Bodner, K., Brimacombe, C., Chenery, E. S., Greiner, A., McLeod, A. M., Penk, S. R., &amp; Soto, J. S. V. (2021). Ten simple rules for tackling your first mathematical models: A guide for graduate students by graduate students. PLOS Computational Biology, 17(1), e1008539.</a></li>
<li><a href="https://jech.bmj.com/content/65/1/87">Mishra, S., Fisman, D. N., &amp; Boily, M.-C. (2011). The ABC of terms used in mathematical models of infectious diseases. Journal of Epidemiology &amp; Community Health, 65(1), 87–94.</a></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><a href="https://doi.org/10.1177/0272989X21990391">James, L. P., Salomon, J. A., Buckee, C. O., &amp; Menzies, N. A. (2021). The Use and Misuse of Mathematical Modeling for Infectious Disease Policymaking: Lessons for the COVID-19 Pandemic. 41(4), 379–385.</a></li>
<li><a href="https://doi.org/10.1056/nejmp2009027">Holmdah, I., &amp; Buckee, C. (2020). Wrong but useful—What COVID-19 epidemiologic models can and cannot tell us. New England Journal of Medicine.</a></li>
<li><a href="https://doi.org/10.1016/j.epidem.2014.08.008">Metcalf, C. J. E. E., Edmunds, W. J., &amp; Lessler, J. (2015). Six challenges in modelling for public health policy. Epidemics, 10(2015), 93–96.</a></li>
<li><a href="https://doi.org/10.1016/j.epidem.2014.09.006">Roberts, M., Andreasen, V., Lloyd, A., &amp; Pellis, L. (2015). Nine challenges for deterministic epidemic models. Epidemics, 10(2015), 49–53.</a></li>
</ul>
</section>
<section class="slide level2">

<h3 id="deriving-and-interpreting-r0">Deriving and Interpreting R0</h3>
<ul>
<li><a href="https://web.stanford.edu/~jhj1/teachingdocs/Jones-on-R0.pdf">Jones, J. H. (2011). Notes On R0. Building, 1–19.</a></li>
<li><a href="https://doi.org/10.1007/BF00178324">Diekmann, O., Heesterbeek, J. A. P., &amp; Metz, J. A. J. (1990). On the definition and the computation of the basic reproduction ratio R0 in models for infectious diseases in heterogeneous populations. Journal of Mathematical Biology, 28(4), 365–382.</a></li>
<li><a href="https://doi.org/10.1098/rsif.2009.0386">Diekmann, O., Heesterbeek, J. A. P., &amp; Roberts, M. G. (2010). The construction of next-generation matrices for compartmental epidemic models. Journal of the Royal Society Interface, 7(47), 873–885.</a></li>
</ul>
</section>
<section class="slide level2 smaller scrollable" id="references">

<h3 id="references">References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-anderson1982directly" class="csl-entry" role="listitem">
Anderson, Roy M, and Robert M May. 1982. <span>“Directly Transmitted Infections Diseases: Control by Vaccination.”</span> <em>Science</em> 215 (4536): 1053–60.
</div>
<div id="ref-Blackwood2018a" class="csl-entry" role="listitem">
Blackwood, Julie C., and Lauren M. Childs. 2018. <span>“An Introduction to Compartmental Modeling for the Budding Infectious Disease Modeler.”</span> <em>Letters in Biomathematics</em> 5 (1): 195–221. <a href="https://doi.org/10.1080/23737867.2018.1509026">https://doi.org/10.1080/23737867.2018.1509026</a>.
</div>
<div id="ref-Box1979" class="csl-entry" role="listitem">
Box, GE. 1979. <span>“All Models Are Wrong, but Some Are Useful.”</span> <em>Robustness in Statistics</em> 202 (1979): 549.
</div>
<div id="ref-diekmann2010construction" class="csl-entry" role="listitem">
Diekmann, Odo, JAP Heesterbeek, and Michael G Roberts. 2010. <span>“The Construction of Next-Generation Matrices for Compartmental Epidemic Models.”</span> <em>Journal of the Royal Society Interface</em> 7 (47): 873–85.
</div>
<div id="ref-diekmann1990definition" class="csl-entry" role="listitem">
Diekmann, Odo, Johan Andre Peter Heesterbeek, and Johan Anton Jacob Metz. 1990. <span>“On the Definition and the Computation of the Basic Reproduction Ratio r 0 in Models for Infectious Diseases in Heterogeneous Populations.”</span> <em>Journal of Mathematical Biology</em> 28: 365–82.
</div>
<div id="ref-guerra2017measlesR0" class="csl-entry" role="listitem">
Guerra, Fiona M, Shelly Bolotin, Gillian Lim, Jane Heffernan, Shelley L Deeks, Ye Li, and Natasha S Crowcroft. 2017. <span>“The Basic Reproduction Number (R0) of Measles: A Systematic Review.”</span> <em>The Lancet Infectious Diseases</em> 17 (12): e420–28.
</div>
<div id="ref-kermack1927contribution" class="csl-entry" role="listitem">
Kermack, William Ogilvy, and Anderson G McKendrick. 1927. <span>“A Contribution to the Mathematical Theory of Epidemics.”</span> <em>Proceedings of the Royal Society of London. Series A, Containing Papers of a Mathematical and Physical Character</em> 115 (772): 700–721.
</div>
</div>
</section>
<section class="slide level2">

<p><img data-src="https://i.creativecommons.org/l/by/4.0/88x31.png"> This presentation is made available through a <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.</p>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-text-resizer/revealjs-text-resizer.js"></script>
  <script src="slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, RevealTextResizer, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
    <script>var lightboxQuarto = GLightbox({"selector":".lightbox","descPosition":"bottom","closeEffect":"zoom","loop":false,"openEffect":"zoom"});
=======
    <script>var lightboxQuarto = GLightbox({"descPosition":"bottom","closeEffect":"zoom","openEffect":"zoom","selector":".lightbox","loop":false});
>>>>>>> d12ec71 (Activate renv)
=======
    <script>var lightboxQuarto = GLightbox({"selector":".lightbox","descPosition":"bottom","openEffect":"zoom","loop":false,"closeEffect":"zoom"});
>>>>>>> 5f85b62 (Add more slides and improve existing slides)
=======
    <script>var lightboxQuarto = GLightbox({"loop":false,"descPosition":"bottom","selector":".lightbox","closeEffect":"zoom","openEffect":"zoom"});
>>>>>>> ca2d4fe (Add more material)
=======
    <script>var lightboxQuarto = GLightbox({"selector":".lightbox","closeEffect":"zoom","loop":false,"openEffect":"zoom","descPosition":"bottom"});
>>>>>>> 75ea54a (Add more material)
=======
    <script>var lightboxQuarto = GLightbox({"openEffect":"zoom","closeEffect":"zoom","loop":false,"descPosition":"bottom","selector":".lightbox"});
>>>>>>> 298b966 (Add more material)
=======
    <script>var lightboxQuarto = GLightbox({"descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox","closeEffect":"zoom"});
>>>>>>> f413a08 (Add more material)
    window.onload = () => {
      lightboxQuarto.on('slide_before_load', (data) => {
        const { slideIndex, slideNode, slideConfig, player, trigger } = data;
        const href = trigger.getAttribute('href');
        if (href !== null) {
          const imgEl = window.document.querySelector(`a[href="${href}"] img`);
          if (imgEl !== null) {
            const srcAttr = imgEl.getAttribute("src");
            if (srcAttr && srcAttr.startsWith("data:")) {
              slideConfig.href = srcAttr;
            }
          }
        } 
      });

      lightboxQuarto.on('slide_after_load', (data) => {
        const { slideIndex, slideNode, slideConfig, player, trigger } = data;
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(slideNode);
=======
    <script>var lightboxQuarto = GLightbox({"openEffect":"zoom","closeEffect":"zoom","descPosition":"bottom","loop":false,"selector":".lightbox"});
=======
    <script>var lightboxQuarto = GLightbox({"descPosition":"bottom","selector":".lightbox","loop":false,"closeEffect":"zoom","openEffect":"zoom"});
>>>>>>> 1668dca (Update slides)
    (function() {
      let previousOnload = window.onload;
      window.onload = () => {
        if (previousOnload) {
          previousOnload();
>>>>>>> 6fbf5e8 (Add more material)
        }
        lightboxQuarto.on('slide_before_load', (data) => {
          const { slideIndex, slideNode, slideConfig, player, trigger } = data;
          const href = trigger.getAttribute('href');
          if (href !== null) {
            const imgEl = window.document.querySelector(`a[href="${href}"] img`);
            if (imgEl !== null) {
              const srcAttr = imgEl.getAttribute("src");
              if (srcAttr && srcAttr.startsWith("data:")) {
                slideConfig.href = srcAttr;
              }
            }
          } 
        });
      
        lightboxQuarto.on('slide_after_load', (data) => {
          const { slideIndex, slideNode, slideConfig, player, trigger } = data;
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(slideNode);
          }
        });
      
      };
      
    })();
              </script>
    

</body></html>